src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -0,0 +1,103 @@
+package nextstep.subway.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
+@Embeddable
+public class Sections {
+	@OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+	private final List<Section> sections = new ArrayList<>();
+
+	public void add(Section section) {
+		validate(section);
+
+		if(isLastSection(section)) {
+			this.sections.add(section);
+			return;
+		}
+
+		if(isFirstSection(section)) {
+			this.sections.add(0, section);
+			return;
+		}
+
+		if(isMiddleSection(section)) {
+			int index = IntStream.range(0, sections.size())
+					.filter(i-> sections.get(i).getDownStation().equals(section.getUpStation()))
+					.findFirst()
+					.orElse(0);
+			if(this.sections.get(index).getDistance() <= section.getDistance()) {
+				throw new IllegalArgumentException();
+			}
+			this.sections.add(index, section);
+		}
```

Comment:
단순히 validate보다는 해당 메서드가 할 행위에 대해 유추할 수 있으면 어떨까요? 
대략적인 수도코드를 작성해보면 이런 흐름은 어떨까 권유드립니다.

```suggestion
		if(addable(section)){
		    SectionAddableType type = SectionAddableType.from(section);
		    type.action(sections, section);
		}
		
		...
	        enum SectionAddableType {
	            PREPEND(condition, action), 
	            APPEND(condition, action), 
	            MIDDLE(condition, action);
	        }
```